lib/Query_Cache.cpp

```diff
@@ -515,8 +515,18 @@ unsigned char* eof_to_ok_packet(QC_entry_t* entry) {
 	// Initialize affected_rows and last_insert_id to zero
 	memset(vp, 0, 2);
 	vp += 2;
+	// Extract warning flags and status from 'EOF_packet'
+	char* eof_packet = entry->value + entry->row_eof_pkt_offset;
+	eof_packet += sizeof(mysql_hdr);
+	// Skip the '0xFE EOF packet header'
+	eof_packet += 1;
+	uint16_t warnings = *reinterpret_cast<uint16_t*>(eof_packet);
+	eof_packet += 2;
+	uint16_t status_flags = *reinterpret_cast<uint16_t*>(eof_packet);
 	// Copy the warning an status flags
-	memcpy(vp, it, 4);
+	*reinterpret_cast<uint16_t*>(vp) = status_flags;
```

Comment:
I think we should avoid unaligned memory access.
This is dangerous on ARM CPUs , see [this example](https://developer.arm.com/documentation/ka003038/latest).
`memcpy()` should be used instead.
I see this code was inspired by previous implementation: I see similar code in `ok_to_eof_packet()` , so maybe we should fix that also in `ok_to_eof_packet()` . Can you fix that too?

Further note: it also do not keep into consideration the endianness of the system, but I think in all ProxySQL code there is the assumption it is running on a little-endianness system.

Reply:
Yes, you are right. I will fix this commit and do another commit for `ok_to_eof_packet()`

Thank you for the review!!